pkg/kyamls/helpers_test.go

```diff
@@ -0,0 +1,69 @@
+package kyamls
+
+import (
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	"testing"
+)
+
+func TestGetLabels(t *testing.T) {
+	path := "test_data/helpers/labelled-secret.yaml"
+	rNode, readErr := yaml.ReadFile(path)
+	require.NoError(t, readErr)
+
+	labels, err := GetLabels(rNode, path)
+	require.NoError(t, err)
+
+	value, _ := labels["gitops/type"]
+	assert.Equal(t, "\"top-secret\"", value)
+}
+
+func TestGetAnnotations(t *testing.T) {
+	path := "test_data/helpers/labelled-secret.yaml"
+	rNode, readErr := yaml.ReadFile(path)
+	require.NoError(t, readErr)
+
+	annotations, err := GetAnnotations(rNode, path)
+	require.NoError(t, err)
+
+	value, _ := annotations["size"]
+	assert.Equal(t, "small", value)
+
+	value, _ = annotations["what"]
+	assert.Equal(t, "\"put.in\"", value)
+}
+
+func TestGetMetadataMap(t *testing.T) {
+	type test struct {
+		path                      string
+		expectedAnnotationsErrMsg string
+		expectedLabelsErrMsg      string
+	}
+
+	tests := []test{
+		{
+			path:                      "test_data/helpers/empty-file.yaml",
+			expectedAnnotationsErrMsg: "failed to get annotations: wrong Node Kind for  expected: MappingNode was : value: {null}",
```

Comment:
I think you can just test if it's an error or not, and that should be fine. Error messages can change, so checking the exact error message will make the tests brittle imo.
You can add a a bool to the test struct named `hasError` or `isError`, and assert if there is an error or not based on the boolean value.

Reply:
Reasonable, will do.

Reply:
Fixed.